modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/transaction/TransactionValidator.scala

```diff
@@ -17,7 +18,10 @@ import eu.timepit.refined.auto._
 
 trait TransactionValidator[F[_]] {
 
-  def validate(signedTransaction: Signed[Transaction]): F[TransactionValidationErrorOr[Signed[Transaction]]]
+  def validate(
+    signedTransaction: Signed[Transaction],
+    ordinal: Option[SnapshotOrdinal]
```

Comment:
I think you should always know the ordinal to validate against. What is the use case when you don't know the ordinal?

Reply:
I guess that the problem is that the only case when ordinal can be `Option` is when `setInitial` and `set` have not been called yet. Afaik transactions can be sent to the node that still hasn't finished the download. Maybe we should reject all the transactions if the snapshot is not set (or to be more specific if the node is in DownloadInProgress state)?

Another solution could be to use `SnapshotOrdinal.MaxValue` as default but it is a dirty one. On one hand it blocks transaction "just in case" but on the other hand it can make false negatives.